src/cart/crt_hg.c

```diff
@@ -655,7 +655,7 @@ crt_get_opx_info_string(char *provider, char *domain, char *ip,
 		D_GOTO(out, rc = -DER_INVAL);
 	}
 
-	strcpy(domain_name, domain);
+	strncpy(domain_name, domain, 9);
 	strtok_r(domain_name, &domain[delimiter], &hfi_str);
```

Comment:
you could use `sizeof(domain_name) - 1` so that if somebody changes the size of `domain_name` that will also get updated.

Reply:
good point, fixed